scanwise/src/main/java/com/appwise/scanner/base/CameraManager.kt

```diff
@@ -15,24 +15,56 @@ import androidx.camera.core.TorchState
 import androidx.camera.lifecycle.ProcessCameraProvider
 import androidx.camera.view.PreviewView
 import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
 import androidx.lifecycle.LifecycleOwner
 import com.appwise.scanner.CameraSearchType
 import com.appwise.scanner.barcode.CodeAnalyzer
+import com.appwise.scanner.managers.PermissionManager
 import com.appwise.scanner.qr.QRAnalyzer
 import com.appwise.scanner.text.TextAnalyzer
 import java.util.concurrent.ExecutionException
 import java.util.concurrent.ExecutorService
 import java.util.concurrent.Executors
 import java.util.concurrent.TimeUnit
 
-class CameraManager(
-    private val context: Context,
-    private val finderView: PreviewView,
-    private val lifecycleOwner: LifecycleOwner,
-    private val targetOverlay: () -> TargetOverlay,
-    // the default for this is the barcode scanner
+class CameraManager {
```

Comment:
I think, this should/could be an `object` as well. Especially when using the `init` functions.